src/core/mac/sub_mac.cpp

```diff
@@ -1013,16 +1012,25 @@ void SubMac::HandleCslTimer(void)
         break;
 
     case kCslSleep:
-        mCslState = kCslSample;
+        mCslSampleTime += periodUs;
 
+#if OPENTHREAD_CONFIG_MAC_SOFTWARE_RX_TIMING_ENABLE
```

Comment:
Define something like `OT_RADIO_CAPS_RECEIVE_TIMING` like `OT_RADIO_CAPS_TRANSMIT_TIMING` as radio capability and replace this macro check with runtime check `SubMac::ShouldHandleReceiveTargetTime` like `SubMac::ShouldHandleTransmitTargetTime`?

Reply:
This was my original idea for the first commit (https://github.com/openthread/openthread/pull/6343/commits/0d92ffb0cd52937527776e681b7fb92a46b18236), but I see two problems:
- Adds unneeded code size.
- Forces all platforms to implement a dummy `otPlatRadioReceiveAt`.

Reply:
I think it is for supporting both NCP mode and RCP mode(posix host + RCP firmware). In RCP mode, `SubMac.cpp` will be included in **both** posix host and RCP firmware. In host side, it will get the radio capability from `SubMac::GetCaps` which combines the real radio capability and the SW capability as the `SubMac` can add SW capability which is missed in radio. If `posix host` finds RCP supports `OT_RADIO_CAPS_RECEIVE_TIMING`, it should not start the receive timer and leave it to RCP. On RCP side, this could be done in `SubMac` layer or radio layer, depends on whether radio has the capability or not in `Get<Radio>().GetCaps()`. If radio supports it, it will leave it to radio and should not start receive timer, if not, it will handle it in `SubMac` layer.

In NCP mode, there will be only one `SubMac` layer and it decides if `SubMac` SW stack should handle the receive timer or not base on Radio's capability.

The slight increase of code size is unavoidable as we share the same `SubMac` code for both posix host build and NCP, RCP firmware build. This is also inline with other radio capabilities like `OT_RADIO_CAPS_ACK_TIMEOUT`, `OT_RADIO_CAPS_TRANSMIT_RETRIES` and `OT_RADIO_CAPS_CSMA_BACKOFF`.

For dummy `otPlatRadioReceiveAt`, I think you can define a weak function in general in `src/core/radio/radio_platform.cpp` to avoid implement it for all platforms like what we have for `otPlatRadioSetMacFrameCounter` or `otPlatRadioGetCslAccuracy`. How do you think?

Reply:
Thanks for the clarification. It makes sense, I'll add the changes.